CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+# Internal Paths for cmake libraries and Setup install and output Directories
+#------------------------------------------------------------------------------
+# This sets where to look for dependent libraries
+set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} ${CMAKE_BINARY_DIR} ${CMAKE_INSTALL_PREFIX})
+# This sets where to look for dependent library's cmake files
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_BINARY_DIR}/lib/cmake ${CMAKE_BINARY_DIR}/share/cmake)
+
+if (NOT CMAKE_INSTALL_RPATH)
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib:${CMAKE_INSTALL_PREFIX}/lib64")
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+endif ()
+#------------------------------------------------------------------------------
+if (NOT DYAD_EXTERNALLY_CONFIGURED)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin CACHE PATH "Single Directory for all Executables.")
+    set(CMAKE_INCLUDE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/include CACHE PATH "Store the headers.")
+    set(EXECUTABLE_OUTPUT_PATH ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib CACHE PATH "Single Directory for all Libraries")
+    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib CACHE PATH "Single Directory for all static libraries.")
+endif ()
+#-----------------------------------------------------------------------------
+# Targets built within this project are exported at Install time for use
+# by other projects.
+#-----------------------------------------------------------------------------
+if (NOT DYAD_EXPORTED_TARGETS)
+    set(DYAD_EXPORTED_TARGETS "dyad-targets")
+endif ()
+#------------------------------------------------------------------------------
+# Build options
+#------------------------------------------------------------------------------
+# option(ENABLE_DYAD_DEBUG "Include debugging prints and logging" OFF) # This is not needed as we have CMAKE_BUILD_TYPE
+# This is verbose, maybe an alternate might help simplify
+#option(BUILD_URPC "Build DYAD's URPC code" OFF)
+#option(ENABLE_PERFFLOW "Build with PerfFlow Aspect support" OFF)
+#option(ENABLE_UCX_DTL "Build DYAD's UCX data transport layer" OFF)
+
+set(DYAD_CONTROL_PLANE "FLUX_RPC" CACHE STRING "Protocol to use for DYAD's Control Plane")
+set_property(CACHE DYAD_CONTROL_PLANE PROPERTY STRINGS FLUX_RPC)
+set(DYAD_DATA_PLANE "FLUX_RPC" CACHE STRING "Protocol to use for DYAD's Data Plane")
+set_property(CACHE DYAD_DATA_PLANE PROPERTY STRINGS FLUX_RPC UCX)
+set(DYAD_PROFILER "NONE" CACHE STRING "Protocol to use for DYAD's Data Plane")
+set_property(CACHE DYAD_PROFILER PROPERTY STRINGS PERFFLOW_ASPECT CALIPER NONE)
+
+#------------------------------------------------------------------------------
+# Build Type based configurations
+#------------------------------------------------------------------------------
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} -std=c99)
```

Comment:
We should just use the consistent standard with c++

Reply:
when i switch the code to c++ we can move that right?

Reply:
FIXED